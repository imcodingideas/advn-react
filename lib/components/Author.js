import React from 'react';

class Author extends React.Component {
  static nameFormatter = (author) => {
    return `${author.lastName}, ${author.firstName}`;
  }
  render() {
    const { author } = this.props;
    return (
      <div>
        <div>{Author.nameFormatter(author)}</div>
        <div>{author.website}</div>
      </div>
    );
  }
}

export default Author;