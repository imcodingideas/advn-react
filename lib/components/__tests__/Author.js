import React from 'react';
import Author from '../Author';

import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(
    <Author author={{
      firstName: 'Joseph',
      lastName: 'Chambers',
      website: 'test.com'
    }} />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});