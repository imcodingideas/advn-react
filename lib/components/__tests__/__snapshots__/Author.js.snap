// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div>
    Chambers, Joseph
  </div>
  <div>
    test.com
  </div>
</div>
`;
